### DISCUSSION TITLE
How to add shortcuts on user "Public" desktop by default?

### DISCUSSION THREAD

**Discussion by imRadwolf** [DustinBrett/daedalOS#230] [Q&A] [answered]
Hello everyone !

Since there's a lack of shortcuts examples in :
./daedalOS/out/Users/Public/Desktop/

I'm struggling to create new ones.

I've gathered reading posts here that the template of a shortcut should be as such :
```
[InternetShortcut]
BaseURL=
Comment=
IconFile=
Type=
URL=
```

But I'm unsure what value to set in the fields.
For example, I'd like to make shortcuts :
* to chromium opening directly a specific web page
* to a specific .pdf file

And other cool stuff that may be possible.
In the only shortcut available on the Desktop, I've managed to make the shortcut open a specific folder by defaut, but I could really use a guide of what's possible in there.

Once I catch the drift I would even volunteer to write the readme for others if that would help !

Thanks for your attention and all your work !

**Comment by DustinBrett** [comment]
Thanks for the post! You are right about that shortcut format but at the moment the Browser app actually doesn't properly process the Url parameter so likely you can't make a shortcut currently to open the browser to a specific url. You could open the PDF app with a Url, but the Url would need to be "local" to daedalOS, and not something via http protocol.

I will add Url parsing to Browser asap.

**Comment by DustinBrett** [comment]
As for what the values mean, `BaseURL` is the name of the app. `URL` points to a local file system path typically, but some apps can understand other URL's like IPFS. `Type` is only for desktop system icons like `My PC` which I wanted to act slightly special (no shortcut icon, sorted first)

**Comment by imRadwolf** [comment, marked as answer]
Okay okay, I get what you're saying about `URL`, not really a problem

About the `BaseURL` part : I've tried multiple things to no avail but it may be because I'm as dumb as they comes...

> As for what the values mean, BaseURL is the name of the app.

I've searched for the program folder :
```
root@marbe # [/home/wwwroot/daedalOS/out/Users/Public/Desktop] >>>>>>>>> : ll ../../../Program\ Files/P
PDF.js/
```

And I tried to then add a `pdf.url` under the path :
```/home/wwwroot/daedalOS/out/Users/Public/Desktop```

And it didn't appeared on the desktop after clearing the session :
![image](https://user-images.githubusercontent.com/16869232/200407206-c0baa50d-3150-46e1-8f00-40398a79e74b.png)

Here are the two tries I made :
```
[InternetShortcut]
BaseURL=PDF
Comment=PDF
IconFile=/home/wwwroot/daedalOS/out/System/Icons/144x144/chromium.png
Type=System
URL=
```
And
```
[InternetShortcut]
BaseURL=PDF.js
Comment=PDF
IconFile=/home/wwwroot/daedalOS/out/System/Icons/144x144/chromium.png
Type=
URL=
```

I don't think I'm searching in the right place for the "name of the app"
Do you happen to have all the right names to use in those shortcuts ?

Thank you once again !

**Reply by DustinBrett** [reply]
You're quite close, I'm glad to help.

You can find a list of all `BaseURL` apps in [directory.ts](https://github.com/DustinBrett/daedalOS/blob/main/contexts/process/directory.ts) as this is the process/app list. You can click on any file and press `Create shortcut` to have it make an example shortcut. For example my resume PDF can make a shortcut using BaseURL PDF like your example:

```
[InternetShortcut]
BaseURL=PDF
IconFile=/System/Icons/pdf.webp
URL=/Users/Public/Documents/Dustin Brett Resume (2022).pdf
```

If you want these shortcut files to persist resets, you'd need to put the files in `/home/wwwroot/daedalOS/public` in the relative folder.

**Reply by imRadwolf** [reply]
I tried a lot of things but it did not work until I ;
1. `npm run build`
2. `npm run export`

Now the real fun can begin :) (well... tomorrow because it's now 1am ._.)

Thank you once again for your support !
